flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -574,4 +609,9 @@ private ContainerLaunchContext createTaskExecutorLaunchContext(Resource resource
 				.put(ENV_FLINK_NODE_ID, host);
 		return taskExecutorLaunchContext;
 	}
+
+	@VisibleForTesting
+	Executor getStartContainerExecutor() {
+		return startContainerExecutor;
+	}
```

Comment:
This looks like a code smell. If we pass in the `ExecutorService`, then there should be no need for a `@VisibleForTesting` method.